sympy/matrices/matrices.py

```diff
@@ -2508,9 +2520,15 @@ def do(x):
                     rows_range = rows_range[:rows]
                 elif cols < rows:
                     cols_range = cols_range[:cols]
-                for i in rows_range:
-                    flat_list.extend(
-                        [cls._sympify(op(i, j)) for j in cols_range])
+
+                if sympify_callback_return:
```

Comment:
recommend
``python
for i in rows_range:
    flat_list.extend([op(i, j) for j in cols_range])
if sympify_callback_return:
    flat_list = cls._sympify(flat_list)
```
And I highly recommend just dropping the option to sympify the input because, as you point out, this is only an `O(n)` step in an `O(n**2)` process.

Reply:
Well, I think that things have to be done like either one of these
1. Sympify first and if things are keeping sympy context, it doesn't necessarily have to go through `o(n**2)` sympifying process
2. Sympify last and it doesn't necessarily have to go through `o(n)` index sympifying process,
3. If we make two flag `False`, sympy types should be created inside the callback function

And the third one has significant slowdown like the previous version than this, because from the mapped function, integer objects should be created `n^2` times, and this was why `Matrix` constructor can take advantage of storing pre-made things in a vector.